### PR TITLE
feat: [DHIS2-15483] assign an user when scheduling an enrollment event

### PR DESCRIPTION
**Pull request by simonadomnisoru**
[DHIS2-15483](https://dhis2.atlassian.net/browse/DHIS2-15483)

**Tech summary**
- Assignee Section into the Sechdule enrollment event tab
- Show the `No results found` message if a searched user does not exist in the system
- store `assignedUser` properties in the `enrollmentDomain` in the redux store

[DHIS2-15483]: https://dhis2.atlassian.net/browse/DHIS2-15483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by eirikhaugstulen — COMMENTED** [review]
Hey @simonadomnisoru,
Just a quick comment, but otherwise very nice!

**Review by JoakimSM — COMMENTED** [review]
When saving the event to the enrollmentDomain in the Redux store, let's make sure we also include displayName, username, firstName and surname like we do when we initially fetch the events. This way it will be easier when we are going to display the assigned user in stages and events later. I think we also use the enrollmentDomain event data directly in the assignee Widget.

File: src/core_modules/capture-core/components/WidgetEventSchedule/Assignee/Assignee.component.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+});
+
+const AssigneePlain = (props: Props) => {
+    const { classes, assignee, ...passOnProps } = props;
+    return (
+        <div className={classes.container}>
+            <div className={classes.label}>{i18n.t('Assignee')}</div>
+            <div className={classes.field}>
+                {/* $FlowFixMe[cannot-spread-inexact] automated comment */}
+                <UserField inputPlaceholderText={i18n.t('Search for user')} value={assignee} {...passOnProps} />
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetEventSchedule/Assignee/Assignee.component.js:32]: Hey @eirikhaugstulen 
I agree that the users will benefit from a `No suggestions`-state (the design team also liked the idea). I added it in the last commits.
Regarding the second point, if the user types very fast, the search is prevented on purpose. Within the UserField.component, the [DebounceField](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-ui/DebounceField/DebounceField.component.js) is used to introduce a delay between search requests. Input debouncing serves the purpose of limiting the frequency of API calls and enhancing overall performance. 
Thanks!

**Comment by simonadomnisoru** [comment]
> When saving the event to the enrollmentDomain in the Redux store, let's make sure we also include displayName, username, firstName and surname like we do when we initially fetch the events. This way it will be easier when we are going to display the assigned user in stages and events later. I think we also use the enrollmentDomain event data directly in the assignee Widget.

Hey @JoakimSM,
Good catch! I added the rest of the `assignedUsper` proprieties into the `enrollmentDomain`. Let me know if you encounter any other issues.
Thanks for the review!

**Review by eirikhaugstulen — APPROVED** [review]
Looks good @simonadomnisoru 👍 Thanks for implementing the changes!

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.41,2.40.2,2.39.4,2.38.6 versions

**Pull request merged**